force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +140,29 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        //Opportunity opp = new Opportunity();
+  
+        //Account testAcc1 = new Account(Name = accountName);
+        //upsert testAcc1;
+
+       List<Account> acc = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];     
+       Id myId;
+       if(acc.size()>0){
+       myId = acc.get(0).Id;
+       }
+       else{
+        Account testAcc1 = new Account(Name = accountName);
+        Insert testAcc1;
+       myId = testAcc1.Id;
+       }
```

Comment:
Excellent, this is the type of thinking we are looking for. There's simpler ways to do this, but I want you to be proud of your solution here because it's readable and it solves the problem.